### ISSUE TITLE
Remove data fetching middleware and lazy load data in handlers

### ISSUE THREAD

**Issue by lipmas** [bastionzero/zli#114]
I dont think we should be continuing to use this patten of pushing data retrieval to this middleware. This increases the number of API calls for every single zli command. We are already starting to see some serious performance issues when running stress tests where many concurrent zli commands are run and each one currently results in 4 API calls just from this middleware layer.

Ideally we should load just the data that is needed in the beginning of each command handler. We can have the data retrieval functions be in common util methods and have each handler load what data it needs in parallel by combining all the methods in a single `await Promise.All()` instead of running all of them in series. 

cc @Nabeelperson If you agree we can open up another issue to refactor the existing code in a separate PR but lets at least not add more middleware API calls here.

_Originally posted by @lipmas in https://github.com/bastionzero/zli/pull/105#discussion_r632160362_

**Comment by lipmas** [comment]
Closed by #495
